# Data Exploration - Hotel Bookings
Group #6: Allyson Vasquez, Alex Miller, Vena Khamvanthong, Mandev Doshi

This notebook explores our dataset to gain deeper insights in order to create meaningful visualizations.

In [44]:
import pandas as pd
import numpy as np
import altair as alt
import streamlit as st

Let's do some exploratory data analysis on our hotel_booking.csv file. This will help us to identify any patterns, relations, or cleaning that needs to be done.

In [48]:
df = pd.read_csv('hotel_booking.csv')

#Looking at the first 10 rows of our dataset
df.head(10)

hotel  is_canceled  lead_time  arrival_date_year arrival_date_month  \
0  Resort Hotel            0        342               2015               July   
1  Resort Hotel            0        737               2015               July   
2  Resort Hotel            0          7               2015               July   
3  Resort Hotel            0         13               2015               July   
4  Resort Hotel            0         14               2015               July   
5  Resort Hotel            0         14               2015               July   
6  Resort Hotel            0          0               2015               July   
7  Resort Hotel            0          9               2015               July   
8  Resort Hotel            1         85               2015               July   
9  Resort Hotel            1         75               2015               July   

   arrival_date_week_number  arrival_date_day_of_month  \
0                        27                          1   
1                        27                          1   
2                        27                          1   
3                        27                          1   
4                        27                          1   
5                        27                          1   
6                        27                          1   
7                        27                          1   
8                        27                          1   
9                        27                          1   

   stays_in_weekend_nights  stays_in_week_nights  adults  ...  customer_type  \
0                        0                     0       2  ...      Transient   
1                        0                     0       2  ...      Transient   
2                        0                     1       1  ...      Transient   
3                        0                     1       1  ...      Transient   
4                        0                     2       2  ...      Transient   
5                        0                     2       2  ...      Transient   
6                        0                     2       2  ...      Transient   
7                        0                     2       2  ...      Transient   
8                        0                     3       2  ...      Transient   
9                        0                     3       2  ...      Transient   

     adr required_car_parking_spaces total_of_special_requests  \
0    0.0                           0                         0   
1    0.0                           0                         0   
2   75.0                           0                         0   
3   75.0                           0                         0   
4   98.0                           0                         1   
5   98.0                           0                         1   
6  107.0                           0                         0   
7  103.0                           0                         1   
8   82.0                           0                         1   
9  105.5                           0                         0   

  reservation_status reservation_status_date              name  \
0          Check-Out              2015-07-01     Ernest Barnes   
1          Check-Out              2015-07-01      Andrea Baker   
2          Check-Out              2015-07-02    Rebecca Parker   
3          Check-Out              2015-07-02      Laura Murray   
4          Check-Out              2015-07-03       Linda Hines   
5          Check-Out              2015-07-03  Jasmine Fletcher   
6          Check-Out              2015-07-03      Dylan Rangel   
7          Check-Out              2015-07-03     William Velez   
8           Canceled              2015-05-06     Steven Murphy   
9           Canceled              2015-04-22     Michael Moore   

                         email  phone-number       credit_card  
0  Ernest.Barnes31@outlook.com  669-792-1661  ************4322  
1       Andrea

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 36 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  object 
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  object 
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 8   stays_in_week_nights            119390 non-null  int64  
 9   adults                          119390 non-null  int64  
 10  children                        119386 non-null  float64
 11  babies                          119390 non-null  int64  
 12  meal            

We can observe that 
- There are 36 columns/attributes.
- There are 119,390 rows/entries.
- Our attributes are objects, integers, or floats.

Let's see if there is any missing data below.

In [50]:
#Checking for missing data
df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_week_nights                   0
adults                                 0
children                               4
babies                                 0
meal                                   0
country                              488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
reserved_room_type                     0
assigned_room_type                     0
booking_changes                        0
deposit_type                           0
agent                              16340
company         

Our dataset does contain missing information, specifically in:
- country
- agent
- company

In [54]:
#Also checking to make sure there aren't any symbols like '?' instead of null values.
#This output also gives us insight into which columns are quantitative and which are categorical
for col in df.columns:
    print('{} : {}'.format(col,df[col].unique()))

hotel : ['Resort Hotel' 'City Hotel']
is_canceled : [0 1]
lead_time : [342 737   7  13  14   0   9  85  75  23  35  68  18  37  12  72 127  78
  48  60  77  99 118  95  96  69  45  40  15  36  43  70  16 107  47 113
  90  50  93  76   3   1  10   5  17  51  71  63  62 101   2  81 368 364
 324  79  21 109 102   4  98  92  26  73 115  86  52  29  30  33  32   8
 100  44  80  97  64  39  34  27  82  94 110 111  84  66 104  28 258 112
  65  67  55  88  54 292  83 105 280 394  24 103 366 249  22  91  11 108
 106  31  87  41 304 117  59  53  58 116  42 321  38  56  49 317   6  57
  19  25 315 123  46  89  61 312 299 130  74 298 119  20 286 136 129 124
 327 131 460 140 114 139 122 137 126 120 128 135 150 143 151 132 125 157
 147 138 156 164 346 159 160 161 333 381 149 154 297 163 314 155 323 340
 356 142 328 144 336 248 302 175 344 382 146 170 166 338 167 310 148 165
 172 171 145 121 178 305 173 152 354 347 158 185 349 183 352 177 200 192
 361 207 174 330 134 350 334 283 153 197 133 241 193 2